src/utils/random.rs

```diff
@@ -35,7 +35,7 @@ type RSamplerFunc<'a, T> = Box<dyn Fn(Box<dyn Iterator<Item = T>>) -> Vec<T> + '
 pub(crate) trait RandomSampler<T: Data>: Send + Sync + Serialize + Deserialize {
     /// Returns a function which returns random samples,
     /// the sampler is thread-safe as the RNG is seeded with random seeds per thread.
-    fn get_sampler(&self) -> RSamplerFunc<T>;
+    fn get_sampler(&self, seed: Option<u64>) -> RSamplerFunc<T>;
```

Comment:
@rajasekarv Is this what you meant by optional seed in Random Sampler?

Reply:
yes